### ISSUE TITLE
Can't switch between English and Korean keyboard with ALT key

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#13128]
Originally reported on https://community.brave.com/t/ubuntu-cant-switch-between-non-english-keyboard-input-mode/185729.

## Steps to Reproduce

1. My Ubuntu is in English
2. My keyboard is Korean (iBusHangul)
3. In Korean keyboard setting, I can switch between English and Korean with ALT key. (Not between generic English and Korean, but WITHIN iBusHangul)
4. But I can’t switch between Korean and English on Brave, while on iBusHangul, using ALT key. (It works on terminal, firefox, and chrome, btw)
5. If I switch the mode on iBusHangul manually by clicking the GUI on top, I can type in Korean, but still can’t switch to English via clicking ALT key.
6. iBusHangul has options to set toggle key. If I use different setup, say Shift+Space, it does work. its Just ALT-R that doesn’t work.

## Actual result:

Doesn't work in Brave, but works fine in Chrome.

## Expected result:

Able to switch between English and Korean input mode on Brave via clicking ALT key. (right ALT key, of course)

## Desktop Brave version:

Ubuntu 18.04.3 LTS (official packages, not Snap)
Version 1.17.75 Chromium: 87.0.4280.88 (Official Build) (64-bit)

**Comment by YerongLi** [comment]
I met a similar issue and my ibus can switch on google-chrome and firefox but it cannot switch input method on brave

**Comment by chiranjeevi-dk** [comment]
I am now facing a same issue for the Japanese IME keyboard (Mozc using IBUS and others too), where the language IME works everywhere even on different browser.

**Comment by EvgenSilkov** [comment]
Same thing can't switch to other language
Ubuntu 22.04.1
❯ brave-browser --version
Brave Browser 135.1.77.95

**Comment by chiranjeevi-dk** [comment]
> I can't switch to other language after opening terminal. The problem seem to be happening because of `setxkbmap -option caps:escape` line in .zshrc. Removing it fixed the problem, remap of the caps is done once with dconf-editor, guess that line was left as a relic of the past Ubuntu 22.04.1 ❯ brave-browser --version Brave Browser 135.1.77.95 Works fine in Manjaro KDE

Thank you for your suggestion! I solved the issue by uninstalling the snap version of Brave and installed the apt version of it. Now everythings works fine.
And the issue you were facing was with the apt version right?

**Comment by EvgenSilkov** [comment]
Yes, tried to install with curl .sh script and with apt, same thing. Never tried snap version.

**Comment by chiranjeevi-dk** [comment]
> Yes, tried to install with curl .sh script and with apt, same thing. Never tried snap version. Oh, yeah, forgot to mention, my keyboard input method right now is XIM (I poked there not sure that I changed it)

Okay, for me i use IME (MOZC for  japanese). In future if i still get the same error after using apt means i'll try to solve it the was you said. But it seems u had issue with terminal itself. For me the bug was only with brave.

**Comment by wawanbreton** [comment]
Hi, quite similar issue for me, main keyboard language used to be `fr` but I switched to `us`. But Brave will only use `fr` layout. I tried to force it in `/etc/default/keyboard` so now `localectl` properly shows `us` as X11 Layout, but that didn't help.
The only difference is that in my case, Chromium behaves the same as Brave, which makes them both unusable... Any idea how I can force it to work, at least with an ugly workaround ?

**Comment by wawanbreton** [comment]
I just found out that I could force the keyboard layout by starting `ibus-setup` and setting the proper keyboard layout here.
However the "Use system keyboard layout" option didn't seem to have any effect, so I could not have an automatic switching following the xkb layout. But at least I have a way to set the keyboard that Brave follows.
